### PR TITLE
♻️ [core] observers + related layout/sizing into #core/dom/layout

### PR DESCRIPTION
**Pull request by rcebulko**
DON'T try to review in the Files tab
Real diff: d56f0023af...dca286ff43
Anything added to core passes typechecking.
Now `core/dom/layout` contains: `index`, `rect`, `page-layout-box`, `intersection`, `intersection-no-root`, `visibility-observer`, `size-observer`

### CODE REVIEW

**Review by honeybadgerdontcare — APPROVED** [review]

**Comment by rcebulko** [comment]
Replacing Justin with Kris since he's out this week

**Review by samouri — APPROVED** [review]

File: src/core/dom/layout/intersection-no-root.js [hunk 1]
```diff
@@ -75,10 +75,10 @@ function getInOb(win) {
  * dedupe to the same promise.
  *
  * @param {!Element} el
- * @return {!Promise<IntersectionObserverEntry>}
+ * @return {!Promise<!IntersectionObserverEntry>}
  */
 export function measureIntersectionNoRoot(el) {
-  if (intersectionDeferreds && intersectionDeferreds.has(el)) {
+  if (intersectionDeferreds?.has(el)) {
```
Comment (kristoferbaxter) [inline src/core/dom/layout/intersection-no-root.js:81]: Idea for a different PR: Wonder if we should apply a sweep PR for this kind of change.

**Review by kristoferbaxter — APPROVED** [review]

File: src/core/dom/layout/intersection-no-root.js (hunk 1 above)
Comment (rcebulko) [reply src/core/dom/layout/intersection-no-root.js:81]: ACK

**Comment by rcebulko** [comment]
Rebasing onto latest to minimize invisible conflicts and setting auto-merge on

**Pull request merged**